protobuf/lib/src/protobuf/coded_buffer_writer.dart

```diff
@@ -337,11 +337,22 @@ class CodedBufferWriter {
         _writeVarint32(value ? 1 : 0);
         break;
       case PbFieldType._BYTES_BIT:
-        _writeBytesNoTag(
-            value is Uint8List ? value : Uint8List.fromList(value));
+        final List<int> bytes = value;
+        if (bytes is Uint8List) {
+          _writeBytesNoTag(bytes);
```

Comment:
If `bytes.length == 0` we still add a splice, etc. Consider adding a `bytes.length == 0` case here inside the `if (is Uint8List)` (as that `.length` load will be inlined if we know it's a `Uint8List`)